edgelet/edgelet-http/src/version.rs

```diff
@@ -1,177 +1,33 @@
 // Copyright (c) Microsoft. All rights reserved.
 
-use futures::{future, Future};
-use hyper::service::{NewService, Service};
-use hyper::{Body, Request, Response};
-use url::form_urlencoded::parse as parse_query;
+use std::fmt;
+use std::str::FromStr;
 
-use error::{Error, ErrorKind};
-use IntoResponse;
+pub const API_VERSION: Version = Version::Version2018_12_30;
 
-pub const API_VERSION: &str = "2018-06-28";
-
-#[derive(Clone)]
-pub struct ApiVersionService<T> {
-    upstream: T,
-}
-
-impl<T> ApiVersionService<T> {
-    pub fn new(upstream: T) -> Self {
-        ApiVersionService { upstream }
-    }
+#[derive(PartialOrd, PartialEq)]
+pub enum Version {
+    Version2018_06_28,
```

Comment:
Put an `_` between `Version` and `2018`

Reply:
I get an error if I add _ between Version and 2018: error: variant `Version_2018_06_28` should have a camel case name such as `Version2018_06_28` 

---
In reply to: [243431431](https://github.com/Azure/iotedge/pull/656#discussion_r243431431) [](ancestors = 243431431)